src/engine/IdTable.h

```diff
@@ -10,23 +10,29 @@
 #include <cstring>
 #include <initializer_list>
 #include <ostream>
+
 #include "../global/Id.h"
+#include "../util/LimitedAllocator.h"
 #include "../util/Log.h"
 
 namespace detail {
 // The actual data storage of the Id Tables, basically a wrapper around a
 // std::vector<Id>
+template <typename Alloc>
```

Comment:
I have changed this, and in general I agree with your reasoning. I chose `Alloc` originally for consistency, because in the C++ STL the allocator parameters to my knowledge are also only called `Alloc`